src/System.Private.CoreLib/shared/System/Globalization/CharUnicodeInfo.cs

```diff
@@ -335,28 +291,20 @@ public static UnicodeCategory GetUnicodeCategory(int codePoint)
         //
         ////////////////////////////////////////////////////////////////////////
 
-        internal static unsafe byte InternalGetCategoryValue(int ch, int offset)
+        internal static byte InternalGetCategoryValue(int ch, int offset)
         {
             Debug.Assert(ch >= 0 && ch <= 0x10ffff, "ch is not in valid Unicode range.");
-            // Get the level 2 item from the highest 12 bit (8 - 19) of ch.
-            ushort index = s_pCategoryLevel1Index[ch >> 8];
-            // Get the level 2 WORD offset from the 4 - 7 bit of ch.  This provides the base offset of the level 3 table.
+            // Get the level 2 item from the highest 11 bits of ch.
+            int index = CategoryLevel1Index[ch >> 9];
+            // Get the level 2 WORD offset from the next 5 bits of ch.  This provides the base offset of the level 3 table.
             // Note that & has the lower precedence than addition, so don't forget the parathesis.
-            index = s_pCategoryLevel1Index[index + ((ch >> 4) & 0x000f)];
+            index = Unsafe.As<byte, ushort>(ref Unsafe.AsRef(in CategoryLevel2Index[(index << 6) + ((ch >> 3) & 0b111110)]));
```

Comment:
This may or may not be aligned at `ushort`. It will cause unpredictable performance changing from build to build in better case and crashes on platforms with strict alignment requirements in worse case.

(Same comment for other uses of `Unsafe.As`.)

Reply:
We need https://github.com/dotnet/corefx/issues/26948 to make this right.

Reply:
There is absolutely no alignment guarantees for static `<PrivateImplementationDetails>` fields?

I can change it to `Unsafe.ReadUnaligned` or should I change it back to a regular `uint[]` array? I would guess that an unaligned read might actually be faster here.

For `InternalGetNumericValue` I'll just change to unaligned read (it isn't 8-byte aligned in the current version also).